sites/partners/__tests__/components/listings/PaperListingForm/UnitGroupForm.test.tsx

```diff
@@ -0,0 +1,329 @@
+    render(
+      <AuthProvider>
+        <FormProviderWrapper>
+          <UnitGroupForm
+            onClose={jest.fn()}
+            onSubmit={jest.fn()}
+            defaultUnitGroup={tempUnitGroup}
+            draft={true}
+            nextId={1}
+          />
+        </FormProviderWrapper>
+      </AuthProvider>
+    )
+
+    const addAmiButton = screen.getByRole("button", { name: /add ami level/i })
+    expect(addAmiButton).toBeInTheDocument()
+
+    await act(() => userEvent.click(addAmiButton))
+
+    const drawerHeader = screen.getByRole("heading", { level: 1, name: /add ami level/i })
+    expect(drawerHeader).toBeInTheDocument()
+
+    const drawerContainer = drawerHeader.parentElement.parentElement
+
+    expect(
+      within(drawerContainer).getByRole("heading", { level: 2, name: /^ami level$/i })
+    ).toBeInTheDocument()
+    expect(within(drawerContainer).getByLabelText(/ami chart/i)).toBeInTheDocument()
+    expect(within(drawerContainer).getByLabelText(/percentage of ami/i)).toBeInTheDocument()
+    expect(within(drawerContainer).getByText(/how is rent determined\?/i)).toBeInTheDocument()
+
+    const fixedAmountOption = within(drawerContainer).getByLabelText(/fixed amount/i)
+    const percentageOption = within(drawerContainer).getByLabelText(/% of income/i)
+    expect(fixedAmountOption).toBeInTheDocument()
+    expect(percentageOption).toBeInTheDocument()
+
+    await act(() => userEvent.click(fixedAmountOption))
+    expect(within(drawerContainer).getByLabelText(/monthly rent/i)).toBeInTheDocument()
+
+    await act(() => userEvent.click(percentageOption))
+    expect(within(drawerContainer).getAllByLabelText(/% of income/i)).toHaveLength(2)
+  })
+
+  it("should render ami charts table", async () => {
+    render(
+      <AuthProvider>
+        <FormProviderWrapper>
+          <UnitGroupForm
+            onClose={jest.fn()}
+            onSubmit={jest.fn()}
+            defaultUnitGroup={tempUnitGroup}
+            draft={true}
+            nextId={1}
+          />
+        </FormProviderWrapper>
+      </AuthProvider>
+    )
+
+    expect(await screen.findAllByText(/mock ami/i))
+
+    const amiTable = screen.getByRole("table")
+    expect(amiTable).toBeInTheDocument()
+
+    const headerAndBody = within(amiTable).getAllByRole("rowgroup")
+    expect(headerAndBody).toHaveLength(2)
+
+    const headerColumns = within(headerAndBody[0]).getAllByRole("columnheader")
+    expect(headerColumns).toHaveLength(5)
+    expect(headerColumns[0]).toHaveTextContent(/ami chart/i)
+    expect(headerColumns[1]).toHaveTextContent(/ami level/i)
+    expect(headerColumns[2]).toHaveTextContent(/how is rent determined\?/i)
+    expect(headerColumns[3]).toHaveTextContent(/monthly rent/i)
+    expect(headerColumns[4]).not.toHaveTextContent()
+
+    const rows = within(headerAndBody[1]).getAllByRole("row")
+    expect(rows).toHaveLength(2)
+
+    const rowOneData = within(rows[0]).getAllByRole("cell")
```

Comment:
nit: for a slightly better readability this can be broken up to have each cell named here and then used in the expect statements

```
    const [amiChart, amiLevel, rentDetermination, monthlyRent, actions] = within(
      rows[0]
    ).getAllByRole("cell")
```